modules/_node-scripts/generate/tsconfig/index.mjs

```diff
@@ -0,0 +1,85 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import path from 'path';
+
+import getProjectDependencies from '../../configuration/getProjectDependencies.mjs';
+import getProjectDescription from '../../configuration/getProjectDescription.mjs';
+import getProjectEntryPoints from '../../configuration/getProjectEntryPoints.mjs';
+import {getProjectDirs, getRootDir, SRC_PATH} from '../../util/constants.mjs';
+import fileExists from '../../util/fileExists.mjs';
+import writeProjectTsconfig from './writeProjectTsconfig.mjs';
+
+export default async function main() {
+	const rootDir = await getRootDir();
+
+	if (path.resolve('.') !== rootDir) {
+		console.error(`
+                                 ⚠️    WARNING    ⚠️
+
+Since generate:tsconfig is a global task, it will be run from the root of liferay-portal even
+though you have invoked it from a project directory.
+`);
+	}
+
+	const [
+		projectDirs,
+	] = await Promise.all([
+		getProjectDirs()
```

Comment:
Does this need to be spread in or in an array? It seems odd to Promise.all with only one item in the array

Reply:
Yeah, that's a leftover from a refactor. I'll change it.